### PR TITLE
kwin: unwrap executable name for resource name

### LINKED ISSUE #118650: Notifications and other popups are treated like "normal" windows on Plasma Wayland
**Describe the bug**
When running Plasma Wayland,  notifications and krunner pop-downs are treated like first-class windows and therefore take focus and appear in the task manager.

This used to be an issue back in the day (https://bugs.kde.org/show_bug.cgi?id=365107) so this is in all likelyhood something we do (wrong) on our side.
    
**To Reproduce**
1. follow the steps in https://github.com/NixOS/nixpkgs/pull/117102 to get wayland running
2. log in to the plasma wayland session
3. wait for a notification or trigger krunner
4. observe the focus steal (for notifications) and appearance in the task manager

**Expected behavior**
Notifications and similar should not be treated as a regular window and therefore not show in task manager or receive focus (krunner should still have focus of course).

**Notify maintainers**
Cc: @NixOS/qt-kde 

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.11, NixOS, 21.05.git.d3ff65233f9M (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210326_dd77f71`
 - channels(root): `""`
 - channels(peter): `""`
 - nixpkgs: `/nix/store/w9zhp284bnx3f1ma6mdbsw008jpnq15v-nixpkgs_unstable`

References:
- https://github.com/NixOS/nixpkgs/pull/117102

### PR DESCRIPTION
**Pull request by CertainLach**
###### Motivation for this change

Meant to fix #118650, but doesn't helps for some reason

app_id is fixed, but notifications still works incorrectly

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by LeoTzahal18 — APPROVED** [review]

**Review by LeoTzahal18 — APPROVED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 139459` at 60679c15 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>37 packages skipped due to time constraints:</summary>
  <ul>
    <li>kdev-php</li>
    <li>kdev-python</li>
    <li>kdevelop-unwrapped</li>
    <li>libsForQt5.plasma-desktop (libsForQt515.plasma-desktop ,plasma5Packages.plasma-desktop)</li>
    <li>libsForQt512.kde-cli-tools</li>
    <li>libsForQt512.kdeplasma-addons</li>
    <li>libsForQt512.khotkeys</li>
    <li>libsForQt512.kinfocenter</li>
    <li>libsForQt512.kmenuedit</li>
    <li>libsForQt512.krohnkite</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>15 packages built successfully:</summary>
  <ul>
    <li>libsForQt5.kde-cli-tools (libsForQt515.kde-cli-tools ,plasma5Packages.kde-cli-tools)</li>
    <li>libsForQt5.kdeplasma-addons (libsForQt515.kdeplasma-addons ,plasma5Packages.kdeplasma-addons)</li>
    <li>libsForQt5.khotkeys (libsForQt515.khotkeys ,plasma5Packages.khotkeys)</li>
    <li>libsForQt5.kinfocenter (libsForQt515.kinfocenter ,plasma5Packages.kinfocenter)</li>
    <li>libsForQt5.kmenuedit (libsForQt515.kmenuedit ,plasma5Packages.kmenuedit)</li>
    <li>libsForQt5.krohnkite (libsForQt515.krohnkite ,plasma5Packages.krohnkite)</li>
    <li>libsForQt5.kwin (libsForQt515.kwin ,plasma5Packages.kwin)</li>
    <li>libsForQt5.kwin-dynamic-workspaces (libsForQt515.kwin-dynamic-workspaces ,plasma5Packages.kwin-dynamic-workspaces)</li>
    <li>libsForQt5.kwin-tiling (libsForQt515.kwin-tiling ,plasma5Packages.kwin-tiling)</li>
    <li>libsForQt5.parachute (libsForQt515.parachute ,plasma5Packages.parachute)</li>
    <li>libsForQt5.plasma-browser-integration (libsForQt515.plasma-browser-integration ,plasma5Packages.plasma-browser-integration)</li>
    <li>libsForQt5.plasma-disks (libsForQt515.plasma-disks ,plasma5Packages.plasma-disks)</li>
    <li>libsForQt5.plasma-workspace (libsForQt515.plasma-workspace ,plasma5Packages.plasma-workspace)</li>
    <li>libsForQt5.powerdevil (libsForQt515.powerdevil ,plasma5Packages.powerdevil)</li>
    <li>libsForQt5.systemsettings (libsForQt515.systemsettings ,plasma5Packages.systemsettings)</li>
  </ul>
</details>

**Review by samueldr — APPROVED** [review]
With the same reservations I had while implementing it initially in #116549 (not a C++ nor Qt expert), approved. Tested with Plasma Mobile.

File: pkgs/desktops/plasma-5/kwin/0001-NixOS-Unwrap-executable-name-for-.desktop-search.patch [hunk 1]
```diff
@@ -8,22 +8,32 @@ Because it's completely bypassing argv0! This looks at the executable
 file in-use according to the kernel!
 
 Wrappers cannot affect the `/proc/.../exe` symlink!
+
+Co-authored-by: Yaroslav Bolyukin <iam@lach.pw>
 ---
```
Comment (SuperSandro2000) [inline pkgs/desktops/plasma-5/kwin/0001-NixOS-Unwrap-executable-name-for-.desktop-search.patch:13]:
```suggestion
```

File: pkgs/desktops/plasma-5/kwin/0001-NixOS-Unwrap-executable-name-for-.desktop-search.patch [hunk 2]
```diff
@@ -8,22 +8,32 @@ Because it's completely bypassing argv0! This looks at the executable
...
- src/service_utils.h | 28 +++++++++++++++++++++++++++-
- 1 file changed, 27 insertions(+), 1 deletion(-)
+ src/nixos_utils.h     | 41 +++++++++++++++++++++++++++++++++++++++++
+ src/service_utils.h   |  4 +++-
+ src/waylandclient.cpp |  5 ++++-
+ 3 files changed, 48 insertions(+), 2 deletions(-)
+ create mode 100644 src/nixos_utils.h
```
Comment (SuperSandro2000) [inline pkgs/desktops/plasma-5/kwin/0001-NixOS-Unwrap-executable-name-for-.desktop-search.patch:19]:
```suggestion
```

File: pkgs/desktops/plasma-5/kwin/0001-NixOS-Unwrap-executable-name-for-.desktop-search.patch (hunk 1 above)
Comment (samueldr) [reply pkgs/desktops/plasma-5/kwin/0001-NixOS-Unwrap-executable-name-for-.desktop-search.patch:13]: What?

File: pkgs/desktops/plasma-5/kwin/0001-NixOS-Unwrap-executable-name-for-.desktop-search.patch (hunk 2 above)
Comment (samueldr) [reply pkgs/desktops/plasma-5/kwin/0001-NixOS-Unwrap-executable-name-for-.desktop-search.patch:19]: What?

**Comment by samueldr** [comment]
@SuperSandro2000 whatever you are doing, whether manual or automatic, it is currently harmful for the completeness of `git format`ed patches.

**Review by ttuegel — APPROVED** [review]

**Pull request merged**